Scalar.Common/Git/GitProcess.cs

```diff
@@ -429,6 +438,38 @@ public Result ForceCheckout(string target)
             return this.InvokeGitInWorkingDirectoryRoot("checkout -f " + target, fetchMissingObjects: true);
         }
 
+        public Result MaintenanceRunTask(MaintenanceTask task, string objectDir)
+        {
+            string taskStr;
+            switch (task)
+            {
+                case MaintenanceTask.CommitGraph:
+                    taskStr = "commit-graph";
+                    break;
+
+                case MaintenanceTask.LooseObjects:
+                    taskStr = "loose-objects";
+                    break;
+
+                case MaintenanceTask.IncrementalRepack:
+                    taskStr = "incremental-repack";
+                    break;
+
+                case MaintenanceTask.Prefetch:
+                    taskStr = "prefetch";
+                    break;
+
+                default:
+                    throw new InvalidOperationException($"Cannot run maintenance task {task}");
+            }
+
+            return this.InvokeGitInWorkingDirectoryRoot(
+                            $"-c pack.window=0 -c pack.depth=0 maintenance run --task={taskStr} --quiet",
```

Comment:
Remind me what/why the `pack.window` and `pack.depth are required for all tasks?

Reply:
You are right, we _shouldn't_ do them for all tasks, just the `loose-objects` task. Otherwise, the `incremental-repack` task will perform much worse. Thanks!